src/models/spaces/charts/chart-data-set.ts

```diff
@@ -73,12 +73,17 @@ export const ChartDataSetModel = types
     fixedMaxA1: types.maybe(types.number),
     fixedMinA2: types.maybe(types.number),
     fixedMaxA2: types.maybe(types.number),
+    // if x data points are not sequential 1,2,3..., and we are setting maxPoints, we need to have an
+    // initial maximum which is not simply the value of maxPoints, which is otherwise the default.
+    initialMaxA1: types.maybe(types.number),
     // expandOnly is used for y-axis scaling. When requesting min/max point values,
     // if this is set the a2 / y axis max returns the max of the full data set, not just the visiblePoints
     expandOnly: false,
     fixedLabelRotation: types.maybe(types.number),
     dataStartIdx: types.maybe(types.number),
-    stack: types.maybe(types.string)
+    stack: types.maybe(types.string),
+    a1AxisLabel: types.maybe(types.string),
```

Comment:
The pattern used earlier is the field description followed by the axis, (thingA1, thingA2), so fixedMaxA1, fixedMinA2, etc. I realize now that the labels I added in the dat-cross-dam repository broke this naming rule, (by going with a1Thing), so this one's my fault. I think it would make sense to fix this to be axisLabelA1, axisLabelA2, to follow the same pattern as the other property names, if that makes sense to you.

Reply:
:+1: updated now